karton/core/config.py

```diff
@@ -34,34 +35,160 @@ class Config(object):
+            if not self.has_section("redis"):
                 raise RuntimeError("Missing Redis configuration")
 
-    @staticmethod
-    def from_dict(config_dict: Dict) -> "Config":
-        config = Config(path=None, check_sections=False)
-        config.config = configparser.ConfigParser()
-        config.config.read_dict(config_dict)
-        return config
+    def set(self, section_name: str, option_name: str, value: Any) -> None:
+        """
+        Sets value in configuration
+        """
+        if section_name not in self._config:
+            self._config[section_name] = {}
+        self._config[section_name][option_name] = value
+
+    def get(
+        self, section_name: str, option_name: str, fallback: Optional[Any] = None
+    ) -> Any:
+        """
+        Gets value from configuration or returns ``fallback`` (None by default)
+        if value was not set.
+        """
+        if not self.has_option(section_name, option_name):
+            return fallback
+        return self._config[section_name][option_name]
+
+    def has_section(self, section_name: str) -> bool:
+        """
+        Checks if configuration section exists
+        """
+        return section_name in self._config
+
+    def has_option(self, section_name: str, option_name: str) -> bool:
+        """
+        Checks if configuration value is set
+        """
+        if section_name not in self._config:
+            return False
+        if option_name not in self._config[section_name]:
+            return False
+        return True
+
+    @overload
+    def getint(self, section_name: str, option_name: str, fallback: int) -> int:
+        ...
+
+    @overload
+    def getint(self, section_name: str, option_name: str) -> Optional[int]:
+        ...
+
+    def getint(
+        self, section_name: str, option_name: str, fallback: Optional[int] = None
+    ) -> Optional[int]:
+        """
+        Gets value from configuration or returns ``fallback`` (None by default)
+        if value was not set. Value is coerced to int type.
+        """
+        value = self.get(section_name, option_name, fallback)
+        if value is None:
+            return None
+        return int(value)
+
+    @overload
+    def getboolean(self, section_name: str, option_name: str, fallback: bool) -> bool:
+        ...
+
+    @overload
+    def getboolean(self, section_name: str, option_name: str) -> Optional[bool]:
+        ...
+
+    def getboolean(
+        self, section_name: str, option_name: str, fallback: Optional[bool] = None
+    ) -> Optional[bool]:
+        """
+        Gets value from configuration or returns ``fallback`` (None by default)
+        if value was not set. Value is coerced to bool type.
+
+        See also:
+        https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.getboolean
+        """
+        value = self.get(section_name, option_name, fallback)
+        if value is None:
+            return None
```

Comment:
We'll have to be careful when checking the return value of `getboolean` since checking if the return value == None is truthy may mislead people